src/main/scala/is/hail/utils/package.scala

```diff
@@ -473,4 +473,21 @@ package object utils extends Logging
     }
     expansion
   }
+
+  def loadFromResource[T](file: String)(reader: (InputStream) => T): T = {
+    val resourceStream = Thread.currentThread().getContextClassLoader.getResourceAsStream(file)
+    assert(resourceStream != null, s"Error while locating file `$file'")
+
+    try {
```

Comment:
I think you should catch/ignore the exception on the close, but just propagate the exception from the reader.  As Tim points out, this an internal error, I don't see what wrapping the exception in a new RuntimeException buys us.

So something like:

```
try { read() }
finally { 
  try { close() } catch { e: Exception => }
}
```

Thoughts?